Mmlu Pro

ysjprojects opened the pull request:
https://huggingface.co/datasets/TIGER-Lab/MMLU-Pro

Added new task: MMLU-Pro

Improvements:
- Higher quality/difficulty compared to original mmlu
- Multiple-choice with 10 options instead of 4

Closes #1947

ysjprojects committed:
initialized mmlu_pro task

ysjprojects committed:
added generative mmlu-pro

ysjprojects committed:
Merge branch 'mmlu-pro' of github.com:ysjprojects/lm-evaluation-harness into mmlu-pro
Resolve conflict.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/EleutherAI/lm-evaluation-harness?pullRequest=1961) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/EleutherAI/lm-evaluation-harness?pullRequest=1961) before we can accept your contribution.<br/>**2** out of **3** committers have signed the CLA.<br/><br/>:white_check_mark: ysjprojects<br/>:white_check_mark: lintangsutawika<br/>:x: Yu Shi Jie<br/><hr/>**Yu Shi Jie** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/EleutherAI/lm-evaluation-harness?pullRequest=1961) it.</sub>

LSinev commented:
You may want add `Closes #1947` to the description to autoclose feature request when this PR will be merged.

LSinev commented on lm_eval/tasks/mmlu_pro/default/_default_template_yaml:
```diff
@@ -0,0 +1,15 @@
+dataset_path: sjyuxyz/MMLU-Pro-with-subset
```
Is this OK to use there some other set than https://huggingface.co/datasets/TIGER-Lab/MMLU-Pro? Or am I missing something?

ysjprojects replied:
Problem with the original dataset is that the categories are not stored as subsets. Instead, there is a 'category' feature. This goes against the mmlu dataset convention and thus cannot integrate well with the existing codebase.

The current set contains no further modifications from the original set except that the categories are split into subsets.

marta-sd replied:
Hi @ysjprojects, unfortunately this dataset differs from the original one. I haven't check everything, but eg. the first example in `dev` was corrupted: question changed from `'The symmetric group $S_n$ has $\n\\factorial{n}$ elements (...)'` to `'The symmetric group $S_n$ has $\nactorial{n}$ elements (...)'`. Also, I've noticed that the number of samples is different: the original one has 12032 rows, while the 'all' variant in the one divided into subsets has 10823.
Do you think it would be possible to fix it?

ysjprojects replied:
Hi, @marta-sd I have pushed some changes to fix the dataset. It should be all good now.

LSinev commented:
I see in the paper https://arxiv.org/abs/2406.01574v3 part about answers extracting

> To extract answers from the model-generated reasoning content in response to the MMLU-Pro dataset, we initially use the regular expression **‘answer is \(?\([A-J]\)?\)’** to match the format specified in the prompt instructions and few-shot examples. If this regex fails to retrieve a valid response, possibly due to formatting deviations by the model, we employ a secondary regex **‘\.*\[aA\]nswer:\s*\([A-J]\)’** for a second attempt to extract the answer. If both of them fail to retrieve a valid response, a **fallback mechanism** is implemented where a **random option** from the answer choices is selected. This ensures consistent answer provision across all evaluations.

It seems, there is no such regexps in this PR, or am I missing them in some nested rules?

ysjprojects commented:
> I see in the paper https://arxiv.org/abs/2406.01574v3 part about answers extracting
> 
> > To extract answers from the model-generated reasoning content in response to the MMLU-Pro dataset, we initially use the regular expression **‘answer is (?([A-J])?)’** to match the format specified in the prompt instructions and few-shot examples. If this regex fails to retrieve a valid response, possibly due to formatting deviations by the model, we employ a secondary regex **‘._[aA]nswer:\s_([A-J])’** for a second attempt to extract the answer. If both of them fail to retrieve a valid response, a **fallback mechanism** is implemented where a **random option** from the answer choices is selected. This ensures consistent answer provision across all evaluations.
> 
> It seems, there is no such regexps in this PR, or am I missing them in some nested rules?

regexps filter is necessary for more complex COT evaluations. COT has yet to be implemented for mmlu-pro, so it is currently not in the PR.

ysjprojects committed:
added cot fewshot for mmlu-pro

Commit:
Initial commit

ysjprojects committed:
updated mmlu-pro to take on 3 splits: test, val, dev

ysjprojects committed:
mmlu-pro: added continuation and flan_cot_zeroshot

ysjprojects committed:
Merge branch 'EleutherAI:main' into mmlu-pro

ysjprojects commented:
Have ported over most mmlu variants to mmlu-pro (continuation, default, flan_cot_fewshot, flan_cot_zeroshotm generative)

Requesting for review @LSinev

lintangsutawika reviewed (commented):
Could you also add a README.md

ysjprojects committed:
added README.md for mmlu_pro

ysjprojects committed:
Merge branch 'mmlu-pro' of github.com:ysjprojects/lm-evaluation-harness into mmlu-pro
sync with remote

ysjprojects commented:
> Could you also add a README.md

done

lintangsutawika committed:
removed

lintangsutawika committed:
update files

lintangsutawika committed:
moved files out, and removed unused versions

lintangsutawika committed:
updated

lintangsutawika commented:
I've made some edits that cut out the tasks variations that aren't in the original paper. 

I was able to reproduce the results for llama-3-instruct on mmlu-pro-math.

Furthermore I'm proposing a minor change in the fewshot sampling process that allows customizing the fewshot example through a `doc_to_text`, `doc_to_target` and `doc_to_choice` that can be customized.

@haileyschoelkopf

Commit:
mmlu_pro:

-changed task 'other' to 'miscellaneous'
there is already a group named 'other'
task and group with the same alias (e.g. mmlu_pro_other_generative) throws an error

-fixed yaml backslash escape for fewshot cot

ysjprojects committed:
changed choices -> options in yaml config to fit dataset schema

Commit:
ONLY FOR DEFAULT: fixed yaml file to use variable number of choices

Commit:
mmlu-pro: fixed doc_to_text/choice/target configs for all variants

Commit:
mmlu-pro: minor fixes

Commit:
Merge branch 'tplx-staging' into 'mmlu-pro'

# Conflicts:
#   README.md

Commit:
Merge branch 'upstream' into 'mmlu-pro'

add tokenizer logs info (#1731)

See merge request shijie.yu/lm-evaluation-harness!4

Commit:
mmlu-pro/default: aligned with mmlu updates

ysjprojects committed:
mmlu-pro: update yaml content in line with mmlu

ysjprojects committed:
mmlu-pro: fixed mislabelling of task (math->chemistry)

ysjprojects committed:
mmlu-pro: fixed yaml formatting

ysjprojects commented:
Hi @lintangsutawika here are some brief updates on the major changes:

1. doc_to_text: variable number of choices

In the MMLU-PRO dataset the number of choices goes up to 10, BUT NOT ALL QUESTIONS HAVE 10 CHOICES. Padding with dummy choices will likely affect performance. Conversely, filtering out questions < 10 choices will cause a 20% drop in dataset size. In both cases it'll go against the canonical implementation.

the easiest approach to solve this is to modify doc_to_text to be able to fetch choices dynamically based on the number of choices, rather than assuming that every question will have exactly 10 choices.

2. There were some breaking changes to the way mmlu configs are formatted in the repo, so I modified mmlu-pro accordingly in line with the changes.

lintangsutawika committed:
add custom fewshot doc_to_text, target, and choice

lintangsutawika committed:
add process for each subtask

lintangsutawika committed:
add process for each subtask

lintangsutawika committed:
pre-commit

lintangsutawika committed:
pre-commit

ysjprojects commented:
> Thanks,
> 
> My main concerns are:
> 
> 1. I would want to keep the attribution to the original dataset. So we should still use TIGER-Lab/MMLU-Pro. I have made some changes that allow us to seperate the dataset based on each category (not yet pushed here)
> 2. There are special cases implemented by the original paper that differentiates the fewshot samples and the options. Mainly the use of `(*)`.
> 3. I prefer to only have a default version, the other variations are not relevant.
>    I have made some modifications from this PR so will make some final adjustments before merging.

Noted with thanks. In that case I'll revert to the original dataset also for my GSM-Plus PR.

lintangsutawika committed:
format

lintangsutawika committed:
merged

lintangsutawika committed:
resolved left out merge

lintangsutawika commented:
@ysjprojects made the PR to your PR here
https://github.com/ysjprojects/lm-evaluation-harness/pull/1

lintangsutawika commented:
Also, could you remove the folders `continuation`, `default`, `flan_cot_fewshot`, `flan_cot_zeroshot`, and `generative`?

ysjprojects committed:
Merge pull request #1 from EleutherAI/mmlu-pro

Mmlu pro

ysjprojects committed:
deleted folders + updated readme

ysjprojects commented:
> Also, could you remove the folders `continuation`, `default`, `flan_cot_fewshot`, `flan_cot_zeroshot`, and `generative`?

Ok done. I've also modified the readme to remove references to the formatted dataset.

lintangsutawika committed:
Update evaluator.py

lintangsutawika committed:
Update evaluator.py

eyuansu62 commented:
@ysjprojects hi, it may need a group config for mmlu_pro.
